File: JavaScript/Intervals/452. Minimum Number of Arrows to Burst Balloons.js
/* 452. Minimum Number of Arrows to Burst Balloons
Difficulty: Medium
Submission Link: https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/submissions/1383579975/ */

function findMinArrowShots(points) {
    points.sort((a, b) => a[1] - b[1]);

    let arrows = 1;
    let prevEnd = points[0][1];

    for (let i = 1; i < points.length; ++i) {
        if (points[i][0] > prevEnd) {
            arrows++;
            prevEnd = points[i][1];
        }
    }

    return arrows;
}

File: Python/Intervals/452. Minimum Number of Arrows to Burst Balloons.py
""" 452. Minimum Number of Arrows to Burst Balloons
Difficulty: Medium
Submission Link: https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/submissions/1383581500/ """

class Solution:
    def findMinArrowShots(self, points: list[list[int]]) -> int:
        if not points:
            return 0
        points.sort(key=lambda x: x[1])
        
        arrows = 1
        end = points[0][1]
        
        for balloon in points:
            start, end_of_balloon = balloon
            if start > end:
                arrows += 1
                end = end_of_balloon
        return arrows
        